src/main/frontend/components/diff.cljs

```diff
@@ -66,75 +67,82 @@
+                   diff (medley/indexed (diff/diff local-content content))
+                   diff? (some (fn [[_idx {:keys [added removed]}]]
+                                 (or added removed))
+                               diff)]
+               [:div.pre-line-white-space.p-2 {:class (if collapse? "hidden")
+                                               :style {:overflow "auto"}}
+                (if edit?
+                  [:div.grid.grid-cols-2.gap-1
+                   (diff-cp diff)
+                   (ui/textarea
+                    {:default-value local-content
+                     :on-change (fn [e]
+                                  (reset! *edit-content (util/evalue e)))})]
+                  (diff-cp diff))
+
+                (cond
+                  edit?
+                  [:div.mt-2
+                   (ui/button "Save"
+                              :on-click
+                              (fn []
+                                (reset! *edit? false)
+                                (let [new-content @*edit-content]
+                                  (file/alter-file repo path new-content
+                                                   {:commit? false
+                                                    :re-render-root? true})
+                                  (swap! state/state
+                                         assoc-in [:github/contents repo remote-oid path] new-content)
+                                  (mark-as-resolved path))))]
+
+                  diff?
+                  [:div.mt-2
+                   (ui/button "Use remote"
+                              :on-click
+                              (fn []
+                                ;; overwrite the file
+                                (if delete?
+                                  (file/remove-file! repo path)
+                                  (file/alter-file repo path content
+                                                   {:commit? false
+                                                    :re-render-root? true}))
+                                (mark-as-resolved path))
+                              :background "green")
+
+                   [:span.pl-2.pr-2 "or"]
+
+                   (ui/button "Keep local"
+                              :on-click
+                              (fn []
+                                ;; overwrite the file
+                                (swap! state/state
+                                       assoc-in [:github/contents repo remote-oid path] local-content)
+                                (mark-as-resolved path))
+                              :background "pink")
+
+                   [:span.pl-2.pr-2 "or"]
+
+                   (ui/button "Edit"
+                              :on-click
+                              (fn []
+                                (reset! *edit? true)))]
+
+                  :else
+                  nil)])))
+         [:div "loading..."]))]))
```

Comment:
When it shows "loading...", will the component refresh itself once the remote ref is ready?

Reply:
Yes, because it subscribed to the content from the remote ref.